demos/nrf52-scanner/src/main.rs

```diff
@@ -0,0 +1,116 @@
+#![no_std]
+#![no_main]
+#![warn(rust_2018_idioms)]
+
+// We need to import this crate explicitly so we have a panic handler
+use panic_halt as _;
+
+// Import the right HAL and PAC
+#[cfg(feature = "52810")]
+use nrf52810_hal as hal;
+
+#[cfg(feature = "52832")]
+use nrf52832_hal as hal;
+
+#[cfg(feature = "52840")]
+use nrf52840_hal as hal;
+
+use {
+    rubble::{
+        beacon::{BeaconScanner, ScanCallback},
+        link::{ad_structure::AdStructure, filter::AllowAll, DeviceAddress, MIN_PDU_BUF},
+        time::{Duration, Timer},
+    },
+    rubble_nrf5x::{
+        radio::{BleRadio, PacketBuffer},
+        timer::BleTimer,
+    },
+};
+
+pub struct BeaconScanCallback;
+
+impl ScanCallback for BeaconScanCallback {
+    fn beacon<'a, I>(&mut self, _addr: DeviceAddress, _data: I)
+    where
+        I: Iterator<Item = AdStructure<'a>>,
+    {
+        // Detected an advertisement frame! Do something with it here.
```

Comment:
Hmm, I think this should at least do *something* with the data, maybe even just an `hprintln!` if that makes sense.

Reply:
We could add a `rprintln!` call through `rtt-target`. It would require connecting with an RTT client though (e.g. by flashing through cargo-embed, which is what I usually do).

Might actually be nice for the demo, since it doesn't require wiring up and configuring UART (and in contrast to semihosting code still runs without a debug probe attached).

Reply:
Since `rtt-target` has a bunch of open soundness issues and is still fairly new, I'd prefer to go with semihosting instead (we already pull that in for a panic handler). Since this is a demo, it won't run without a debugger anyways (and if it did, it wouldn't output anything regardless).